game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/fire/RollNormal.java

```diff
@@ -0,0 +1,30 @@
+package games.strategy.triplea.delegate.battle.steps.fire;
+
+import static games.strategy.triplea.delegate.battle.BattleState.Side.DEFENSE;
+import static games.strategy.triplea.delegate.battle.BattleState.UnitBattleFilter.ACTIVE;
+
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.delegate.DiceRoll;
+import java.util.function.BiFunction;
+
+/** Rolls dice for normal (basically, anything that isn't AA) dice requests */
+public class RollNormal implements BiFunction<IDelegateBridge, RollDice, DiceRoll> {
```

Comment:
Is RollNormal an equivalent thing to "RollGeneralCombat" ?
I wonder if we are mixing "normal" and "general" as synonymous terms.

Reply:
No, normal and general are not synonymous.  Though I'm open to new names.  General is for a phase in the combat.  But normal is for the type of unit.  You have AA/Targeted dice requests which are rolled during the AA phase.  And then you have normal dice requests which rolled in all the other phases (bombard, first strike, and general).  The use of normal comes from `TotalPowerAndTotalRolls`.  It uses "normal" and "aa" there in that manner and I've been following it here since I didn't think a class called `Roll` would be ok.

Reply:
Perhaps then `MainBattleRoll` as a more descriptive name?

In general, "general", "normal", and "standard" are context dependent descriptors and maintainers and other developers simply won't have that context without going through some work. Saving any amount of work to make code more understandable leads to more maintainable code.

Other suggestions: `MainRoll`, `StandardRoll`

Reply:
`Main` sounds good.  I renamed it to `RollMainDice`.